tests/testAdminFeatures.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * Contains Class TestAdminFeatures.
+ *
+ * @package WP-Auth0
+ *
+ * @since 3.8.1
+ */
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class TestAdminFeatures.
+ */
+class TestAdminFeatures extends TestCase {
+
+	use HttpHelpers;
+
+	use RedirectHelpers;
+
+	use SetUpTestDb;
+
+	/**
+	 * Instance of WP_Auth0_Options.
+	 *
+	 * @var WP_Auth0_Options
+	 */
+	public static $opts;
+
+	/**
+	 * WP_Auth0_ErrorLog instance.
+	 *
+	 * @var WP_Auth0_ErrorLog
+	 */
+	protected static $error_log;
+
+	/**
+	 * Runs once before all tests.
+	 */
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+		self::$opts      = WP_Auth0_Options::Instance();
+		self::$error_log = new WP_Auth0_ErrorLog();
+	}
+
+	/**
+	 * Test that the validation does not run if the current value matches the old value.
+	 */
+	public function testThatInputIsNotValidatedIfItIsNotChanged() {
+		$admin_features = new WP_Auth0_Admin_Features( self::$opts );
+
+		$old_input = [ 'password_policy' => 'fair' ];
+		$new_input = [ 'password_policy' => 'fair' ];
+		$result    = $admin_features->security_validation( $old_input, $new_input );
+
+		$this->assertEquals( $old_input, $result );
+
+		global $wp_settings_errors;
+		$this->assertEmpty( $wp_settings_errors );
+	}
+
+	/**
+	 * Test the connection search request structure.
+	 */
+	public function testThatConnectionsAreSearchedDuringValidation() {
+		$this->startHttpHalting();
+
+		$test_domain    = 'test-wp.auth0.com';
+		$test_token     = implode( '.', [ uniqid(), uniqid(), uniqid() ] );
+		$admin_features = new WP_Auth0_Admin_Features( self::$opts );
+
+		$old_input = [ 'password_policy' => 'fair' ];
+		$new_input = [
+			'password_policy' => 'good',
+			'domain'          => $test_domain,
+			'auth0_app_token' => $test_token,
+		];
+
+		$caught_request = [];
+		try {
+			$admin_features->security_validation( $old_input, $new_input );
+		} catch ( Exception $e ) {
```

Comment:
why would the above call throw an exception?

Reply:
I'm forcing it to with `$this->startHttpHalting()` above